DuckDuckGo/AutocompleteViewController.swift

```diff
@@ -202,3 +206,35 @@ extension AutocompleteViewController: Themable {
         tableView.reloadData()
     }
 }
+
+extension AutocompleteViewController {
+ 
+    func keyboardMoveSelectionDown() {
+        guard !suggestions.isEmpty else { return }
+        selectedItem = (selectedItem + 1 >= itemCount()) ? 0 : selectedItem + 1
+        delegate?.autocomplete(pressedPlusButtonForSuggestion: suggestions[selectedItem].suggestion)
+        tableView.reloadData()
+    }
+
+    func keyboardMoveSelectionUp() {
+        guard !suggestions.isEmpty else { return }
+        selectedItem = (selectedItem - 1 < 0) ? itemCount() - 1 : selectedItem - 1
+        delegate?.autocomplete(pressedPlusButtonForSuggestion: suggestions[selectedItem].suggestion)
+        tableView.reloadData()
+    }
+    
+    func keyboardEscape() {
+        delegate?.autocompleteWasDismissed()
+    }
```

Comment:
Maybe we should clear 'selectedItem' when user types something?

Example:
1. Type 'what'.
2. Navigate autocomplete to 'what is my ip'.
3. Delete 'ip' - selection points to new result, but pressing 'enter' in this scenario is not anymore related to selection.

Reply:
Done, thanks